packages/go_router/lib/src/delegate.dart

```diff
@@ -86,38 +145,51 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
   /// See also:
   /// * [push] which pushes the given location onto the page stack.
   void replace(RouteMatch match) {
-    _matches.matches.last = match;
+    _matchList.matches.last = match;
     notifyListeners();
   }
 
   /// For internal use; visible for testing only.
   @visibleForTesting
-  RouteMatchList get matches => _matches;
+  RouteMatchList get matches => _matchList;
 
   /// For use by the Router architecture as part of the RouterDelegate.
   @override
-  GlobalKey<NavigatorState> get navigatorKey => _key;
+  GlobalKey<NavigatorState> get navigatorKey => _configuration.navigatorKey;
 
   /// For use by the Router architecture as part of the RouterDelegate.
   @override
-  RouteMatchList get currentConfiguration => _matches;
+  RouteMatchList get currentConfiguration => _matchList;
 
   /// For use by the Router architecture as part of the RouterDelegate.
   @override
-  Widget build(BuildContext context) => builder.build(
-        context,
-        _matches,
-        pop,
-        navigatorKey,
-        routerNeglect,
-      );
+  Widget build(BuildContext context) {
+    return builder.build(
+      context,
+      _matchList,
+      pop,
+      routerNeglect,
+    );
+  }
 
   /// For use by the Router architecture as part of the RouterDelegate.
   @override
   Future<void> setNewRoutePath(RouteMatchList configuration) {
-    _matches = configuration;
+    _matchList = configuration;
     // Use [SynchronousFuture] so that the initial url is processed
     // synchronously and remove unwanted initial animations on deep-linking
     return SynchronousFuture<void>(null);
   }
 }
+
+/// Thrown when [GoRouter] is used incorrectly.
+class GoError extends Error {
```

Comment:
This is a weird place for this class, maybe put into its own file?

Reply:
9d109db391d0ee06647a9af63a3a50141545466c